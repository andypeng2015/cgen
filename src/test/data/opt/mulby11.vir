module foo

export function w $foo(w %x) {
@start:
  %x = mul.w %x, 11_w
  ret %x
}
