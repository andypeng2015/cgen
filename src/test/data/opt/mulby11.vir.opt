module foo

export function w $foo(w %x) {
@2:
  %0 = lsl.w %x, 0x3_w ; @4
  %1 = lsl.w %x, 0x1_w ; @5
  %2 = add.w %1, %x ; @6
  %3 = add.w %0, %2 ; @7
  ret %3
}
