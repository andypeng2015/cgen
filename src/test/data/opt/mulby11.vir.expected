module foo

export function w $foo(w %x) {
@0:
  %0 = lsl.w %x, 0x3_w ; @2
  %1 = lsl.w %x, 0x1_w ; @3
  %2 = add.w %1, %x ; @4
  %3 = add.w %0, %2 ; @5
  ret %3
}
